torchaudio_contrib/functional.py

```diff
@@ -1,106 +1,133 @@
+    """
+    Wraps _stft setting default values and correct window device.
+    See torchaudio_contrib.STFT for more details.
+    """
+    n_fft, hop_length, window = stft_defaults(
+        n_fft, hop_length, len_win, window)
+    return _stft(x, n_fft=n_fft, hop_length=hop_length,
+                 window=window.to(x.device), pad=pad,
+                 pad_mode=pad_mode, **kwargs)
 
 
-def spectrogram(sig, n_fft=2048, hop=None, window=None, power=1.0, **kwargs):
+def complex_norm(x, power=1.0):
+    """
+    Normalize complex tensor.
     """
-    returns magnitude of spectrogram
+    return x.pow(2).sum(-1).pow(power / 2.0)
 
-    input -> (batch, channel, time) or (channel, time)
-    output -> (batch, channel, freq, hop) or (channel, freq, hop)
+
+def spectrogram(x, n_fft=2048, hop_length=None, len_win=None,
+                window=None, pad=0, pad_mode="reflect", power=1., **kwargs):
+    """
+    Compute the spectrogram of a given signal.
+    See torchaudio_contrib.Spectrogram for more details.
     """
-    stft = STFT(sig, n_fft, hop=hop, window=window, **kwargs)
-    if power is None:
-        return stft
-    return stft.pow(2).sum(-1).pow(power / 2.0)
+    return complex_norm(stft(x, n_fft, hop_length, len_win,
+                             window, pad, pad_mode, **kwargs), power=power)
 
 
 def _hertz_to_mel(f):
-    '''
+    """
     Converting frequency into mel values using HTK formula
-    '''
+    """
     return 2595. * torch.log10(torch.tensor(1.) + (f / 700.))
 
 
 def _mel_to_hertz(mel):
-    '''
+    """
     Converting mel values into frequency using HTK formula
-    '''
+    """
     return 700. * (10**(mel / 2595.) - 1.)
 
 
-def melspectrogram(sig, n_mels=128, sr=44100, f_min=0.0, f_max=None, *args, **kwargs):
+def create_mel_filter(n_mels, sr, f_min, f_max, n_stft):
     """
-    returns Melspectrogram
-    """
-    spec_amp = spectrogram(*args, **kwargs)
-    mel_fb, _ = create_mel_filter(spec_amp.size(-1), sr, n_mels, **kwargs)
-    mel_spec_amp = torch.matmul(spec_amp, mel_fb)
-    return mel_spec_amp
-
-
-def create_mel_filter(n_stft, sr, n_mels=128, f_min=0.0, f_max=None):
-    '''
-    Creates filter matrix to transform fft frequency bins into mel frequency bins.
+    Creates filter matrix to transform fft frequency bins
+    into mel frequency bins.
     Equivalent to librosa.filters.mel(sr, n_fft, htk=True, norm=None).
 
-    Output Tensor shape -> (n_mels, n_stft)
-    '''
+    Args:
+        n_mels (int): number of mel bins.
+        sr (int): sample rate of audio signal.
+        f_max (float, optional): maximum frequency.
+        f_min (float): minimum frequency.
+        n_stft (int, optional): number of filter banks from stft.
+
+    Returns:
+        fb (Tensor): (n_stft, n_mels)
+    """
     # Convert to find mel lower/upper bounds
     f_max = f_max if f_max else sr // 2
-    m_min = 0. if f_min == 0 else _hertz_to_mel(f_min)
+    n_stft = n_stft if n_stft else 1025
```

Comment:
No?

Reply:
Sorry what do you mean? I'll put the default value of `n_stft` as a keyword arg.

Reply:
Sorry, that comment was a bit short ;) Yes, I meant declaring it as a required argument above and then using some undocumented default value in the code. I don't think there should be a default value.